.ci/build-directories.cake

```diff
@@ -0,0 +1,366 @@
+	if (string.IsNullOrWhiteSpace(GIT_COMMIT)) {
+		Information("Git commit was not specified, trying to determine one...");
+		try {
+			GIT_COMMIT = GetGitOutput("rev-parse --verify HEAD")[0];
+		} catch(Exception ex) {
+			throw new Exception("Unable to determine current git hash.", ex);
+		}
+		Information("Found git commit: {0}", GIT_COMMIT);
+	}
+
+	string gitArgs;
+	if (!string.IsNullOrWhiteSpace(GIT_PREVIOUS_COMMIT)) {
+		Information("Previous git specified...");
+		gitArgs = $"--no-pager diff --name-only {GIT_PREVIOUS_COMMIT} {GIT_COMMIT}";
+	} else {
+		Information("No previous git specified, using the changes in the current commit...");
+		gitArgs = $"--no-pager show --pretty=\"format:\" --name-only {GIT_COMMIT}";
+	}
+
+	// Get all the changed files between these commits
+	var changedFilesRaw = GetGitOutput(gitArgs);
+	var changedFiles = new FilePathCollection(changedFilesRaw.Select(f =>(FilePath)f));
+
+	Debug("Found {0} changed file(s):" + Environment.NewLine +
+		" - " + string.Join(Environment.NewLine + " - ", changedFilesRaw),
+		changedFilesRaw.Length);
+
+	// Determine which group each file belongs to
+	var matchedFiles = new Dictionary<DirectoryPath, FilePathCollection>();
+	foreach (var buildGroup in BUILD_GROUPS) {
+		foreach (var file in changedFiles) {
+			var relative = MakeAbsolute(buildGroup).GetRelativePath(MakeAbsolute(file));
+			if (relative.FullPath.StartsWith(".."))
+				continue;
+
+			// Determine if Cocoapods need updating
+			if (IsRunningOnMac() && podRepoUpdate == PodRepoUpdate.NotRequired && POD_UPDATE_PREFIXES.Any(p => buildGroup.FullPath.StartsWith(p)))
+				podRepoUpdate = PodRepoUpdate.Required;
+
+			if (!matchedFiles.TryGetValue(buildGroup, out var match))
+				matchedFiles[buildGroup] = match = new FilePathCollection();
+			if (!match.Contains(file))
+				match.Add(file);
+			if (!groupsToBuild.Contains(buildGroup))
+				groupsToBuild.Add(buildGroup);
+		}
+	}
+
+	Information("");
+	Information("Found {0} changed file(s) that will trigger {1} build(s):",
+		matchedFiles.Values.Sum(m => m.Count),
+		matchedFiles.Keys.Count);
+	foreach (var match in matchedFiles) {
+		Information($" - {{0}}", match.Key);
+		Information("    - " + string.Join(Environment.NewLine + "    - ", match.Value));
+	}
+
+	var extraFiles = changedFiles - new FilePathCollection(matchedFiles.SelectMany(m => m.Value));
+	if (extraFiles.Count > 0) {
+		Information("Found {0} changed file(s) that did not match any item in the manifest:" + Environment.NewLine +
+			" - " + string.Join(Environment.NewLine + " - ", extraFiles),
+			extraFiles.Count);
+	}
+	Information("");
+}
+
+// If a filter was specified, then use it
+if (BUILD_NAMES.Length > 0) {
+	groupsToBuild = groupsToBuild
+		.Where(bg => BUILD_NAMES.Any(n =>
+			// match exact "Android/Kotlin"
+			bg.FullPath.Equals(n, StringComparison.OrdinalIgnoreCase) ||
+			// match partial "Kotlin"
+			bg.FullPath.Replace("\\", "/").EndsWith("/" + n, StringComparison.OrdinalIgnoreCase)))
```

Comment:
Instead of requireing the full directory path, such as "Android/Kotlin", you can also use the simple name of "Kotlin". This is to help with simplifying tags and other things. If there is any ambiguity, then we can fall back to the full name.